### ISSUE TITLE
fails to compile `cargo test` on Mac OS (utmp format)

### ISSUE THREAD

**Issue by jtmoon79** [jtmoon79/super-speedy-syslog-searcher#217]
### Problem

`cargo test` fails to compile on Mac OS.  Here is a run on Mac OS 12.

```
$ cargo test
...
   Compiling super_speedy_syslog_searcher v0.6.68 (/Users/runner/work/super-speedy-syslog-searcher/super-speedy-syslog-searcher)
error[E0308]: mismatched types
    --> src/tests/common.rs:2338:43
     |
2338 |   pub const UTMPX_BUFFER1: [u8; UTMPX_SZ] = [
     |  _______________________________--------____^
     | |                               |
     | |                               help: consider specifying the actual array length: `384`
2339 | |     0x05, 0x00, 0x02, 0x00, 0xb4, 0xa3, 0x00, 0x00, b'p', b't',
2340 | |     b's', b'/', b'1', 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
2341 | |     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
...    |
2377 | |     0x00, 0x00, 0x00, 0x00,
2378 | | ];
     | |_^ expected an array with a fixed size of 640 elements, found one with 384 elements

error[E0308]: mismatched types
    --> src/tests/common.rs:2393:43
     |
2393 |   pub const UTMPX_BUFFER2: [u8; UTMPX_SZ] = [
     |  _______________________________--------____^
     | |                               |
     | |                               help: consider specifying the actual array length: `384`
2394 | |     0x07, 0x00, 0x00, 0x00, 0xb4, 0x33, 0x00, 0x00, 0x70, 0x74,
2395 | |     0x73, 0x2f, 0x30, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
2396 | |     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
...    |
2432 | |     0x00, 0x00, 0x00, 0x00,
2433 | | ];
     | |_^ expected an array with a fixed size of 640 elements, found one with 384 elements

error[E0308]: mismatched types
    --> src/tests/common.rs:2448:43
     |
2448 |   pub const UTMPX_BUFFER3: [u8; UTMPX_SZ] = [
     |  _______________________________--------____^
     | |                               |
     | |                               help: consider specifying the actual array length: `384`
2449 | |     0x07, 0x00, 0x00, 0x00, 0xb4, 0x33, 0x00, 0x00, 0x70, 0x74,
2450 | |     0x73, 0x2f, 0x30, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
2451 | |     0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
...    |
2487 | |     0x00, 0x00, 0x00, 0x00,
2488 | | ];
     | |_^ expected an array with a fixed size of 640 elements, found one with 384 elements

For more information about this error, try `rustc --explain E0308`.
error: could not compile `super_speedy_syslog_searcher` (lib test) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: command `/Users/runner/.rustup/toolchains/stable-x86_64-apple-darwin/bin/cargo test --no-run --message-format json-render-diagnostics --lib --bin s4` exited with code 101
Error: Process completed with exit code 101.
```

Mac OS [discontinued use of utmp format files sometime before Mac OS 11](https://discussions.apple.com/thread/4786112). The libraries are still available after the discontinuation to allow parsing those files from an older system (presumably).

On [Mac OS, a _utmpx_ file consists of](https://github.com/libyal/dtformats/blob/55007dcac48efff42c497e739208ebfb88e4048d/documentation/Utmp%20login%20records%20format.asciidoc):
> The Mac OS X 10.5 utmpx file header is 628 bytes in size and consists of:
> ...
> The Mac OS X 10.5 utmpx record is 628 bytes in size and consists of:
> ...

whereas, on Linux:
> The Linux libc6 utmp recordd is 384 bytes in size and consists of:

### Environment:

 - OS: Mac OS 11, Mac OS 12.6
 - s4 version: after 0.6.68 (db732307106adf4f499411bae742bbbc0d105d25)

### Related

This is another fallout of using `uapi` crate, see meta Issue _refactor utmp/utmpx parsing_ #171.

**Comment by jtmoon79** [comment]
Some workaround code was added in 2c8bf0f `job_test_macos`. The correct fix is #171

**Comment by jtmoon79** [comment]
Built on Mac OS 12 using Github actions https://github.com/jtmoon79/super-speedy-syslog-searcher/actions/runs/8386892984
